### PR TITLE
Rename Property Plugin from PropertyRequiresPlugin to ConfigPropertyPlugin

### PR DESCRIPTION
**Pull request by SentryMan**
We use it for more than the `@RequiresX` annotations. Also I don't like the name.

### CODE REVIEW

File: inject-generator/pom.xml [hunk 1]
```diff
@@ -37,12 +37,6 @@
       <scope>provided</scope>
     </dependency>
 
-    <dependency>
```
Comment (rob-bygrave) [inline inject-generator/pom.xml:40]: Looks unrelated?

Comment (SentryMan) [reply inject-generator/pom.xml:40]: Hmm, thought it came about when I pulled from master and had conflicts

**Comment by rbygrave** [comment]
I'm pondering if we just go for a simple breaking change here.  As in, likely very very few people have implemented this interface (it hasn't been around for a long time etc).  So I think it might be acceptable and simple and less messy to just go with a simple rename and breaking change.  Pondering ...

**Comment by SentryMan** [comment]
> I'm pondering if we just go for a simple breaking change here.

It'll break every single one of the other avaje plugins if we do a clean break. Like the Jsonb Inject plugin for example
